src/components/sidebar/SafeListItem/index.tsx

```diff
@@ -48,7 +49,8 @@ const SafeListItem = ({
   const isCurrentSafe = chainId === currChainId && sameAddress(safeAddress, address)
   const name = allAddressBooks[chainId]?.[address]
   const shortName = chain?.shortName || ''
-  const { totalQueued, totalToSign } = usePendingActions(chainId, isAdded ? address : undefined)
+  const isVisible = useIntersectionObserver(safeRef)
+  const { totalQueued, totalToSign } = usePendingActions(chainId, isVisible, isAdded ? address : undefined)
```

Comment:
```suggestion
  const { totalQueued, totalToSign } = usePendingActions(chainId, isAdded && isVisible ? address : undefined)
```
Since address is optional already...

Reply:
Nice suggestion! This also the previous comment is also addressed 👍

Reply:
No, it won't. It will change from false to true and back as you scroll, triggering loading every time.

Reply:
I've cached the loadedQueue to avoid issuing request on scrolling back and forth